packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -1017,14 +1029,27 @@ function completeWork(
       return null;
     }
     case SuspenseComponent: {
-      popSuspenseContext(workInProgress);
       const nextState: null | SuspenseState = workInProgress.memoizedState;
-
       if (enableSuspenseServerRenderer) {
+        if (
+          enableClientRenderFallbackOnHydrationMismatch &&
```

Comment:
Should this be gated on "concurrent mode"? Because that's the case that would've thrown and therefore attempted to be recovered.